docs/website/configuration.rst

```diff
@@ -0,0 +1,120 @@
+Preview GluonNLP Website Locally
+-----------------------------------------------------------------
+
+The GluonNLP docs website is at `release branch <https://gluon-nlp.mxnet.io>`__, or `master branch <https://gluon-nlp.mxnet.io/master/index.html>`__. Its source code is at `gluon-nlp <https://github.com/dmlc/gluon-nlp>`__.
+
+Currently the GluonNLP website is constructed from the source code via CI automatically. Here I will share:
+
+- the structure of files used for the website, and
+- how to make changes to the website and preview the website
+
+Website Structure
+~~~~~~~~~~~~~~~~~
+
+Currently the docs part contain four sections: Model Zoo, Examples, API and Community. It should be noted that the model zoo is a link redirecting to the ``scripts`` folder in the parent folder. The other three folders are used exclusively by the docs website. Also, three different sections use ``rst``, ``py``, ``md`` files and their composition for compiling - they are inconsistent. So when you work on different sections of the docs website, you should  pay attention to handle the different sections with care.
+
+The main structure, the index file of the entire website, is written in ``rst`` format. It calls the index file of each different section separately. Before compiling the website, you should be aware that:
+
+- ``rst`` files are static files, they are directly displayed to the website with further styles;
+- ``md`` files are script files, the python scripts in these files will be executed and then stored into ``ipynb`` files before converting ``ipynb`` files into website files.
+
+Or more specifically, the files in the examples folder will be further executed and converted into intermediate files before writing to the final HTML files, while those in other folders don’t need further conversion or computation.
+
+Environment Configuration Instruction
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Next, I will give a step by step instruction on how to compile this website from scratch.
+
+1. Converting md files into ipynb files
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1.1 Using existing ipynb from the website
+:::::::::::::::::::::::::::::::::::::::::
+
+If you don’t modify the examples section, what you need to do here is not converting the md files manually(which is time-consuming), but to get the ipynb files that are already ready for use. They can be found by clicking on the “DOWNLOAD THIS TUTORIAL” button on each page of the examples, such as `the word embedding tutorial <https://gluon-nlp.mxnet.io/master/examples/word_embedding/word_embedding.html>`__. After downloading these files, please put them in the same folder along with each md file. And then *delete* the md files. After these steps, you can skip the remaining of this section and go to the next section.
```

Comment:
This requires a lot of manual steps. Could you modify the setup instead in such a way, that it is very easy to build the website without running the notebooks? Ie the website would then just contain the notebooks without their respective outputs. I think we discussed the required steps a few weeks ago. You need to modify `docs/md2ipynb.py` file and introduce a setting there. That could be triggered by an environment variable.